src/main/java/com/microfocus/application/automation/tools/octane/CIJenkinsServicesImpl.java

```diff
@@ -604,232 +647,257 @@ private PipelineNode createPipelineNode(String name, Job job, boolean includePar
-		synchronized (skipOctaneAllowedStoragePrintLock) {
-			//do allowPrint only once
-			allowPrint = !skipOctaneAllowedStoragePrint;
-			skipOctaneAllowedStoragePrint = true;
-		}
-
-		File folder;
-		// jenkins.xml
-		//  <arguments>-Xrs -Xmx256m -octaneAllowedStorage=userContentTemp -Dhudson.lifecycle=hudson.lifecycle.WindowsServiceLifecycle -jar "%BASE%\jenkins.war"
-		String prop = System.getProperty("octaneAllowedStorage");
-		if (StringUtils.isNotEmpty(prop)) {
-			folder = new File(prop);
-			if (!folder.isAbsolute()) {
-				folder = new File(jenkins.getRootDir(), prop);
-			}
-			if (allowPrint) {
-				systemLogger.info("octaneAllowedStorage : " + folder.getAbsolutePath());
-				//validate that folder exist
-				if (!folder.exists() && !folder.mkdirs()) {
-					systemLogger.warning("Failed to create octaneAllowedStorage : " + folder.getAbsolutePath() + ". Create this folder and restart Jenkins.");
-				}
-			}
-		} else {
-			folder = new File(jenkins.getRootDir(), "userContent");
-		}
-		return folder;
-	}
-
-	public static CIServerInfo getJenkinsServerInfo() {
-		CIServerInfo result = dtoFactory.newDTO(CIServerInfo.class);
-		String serverUrl = Jenkins.get().getRootUrl();
-		if (serverUrl != null && serverUrl.endsWith("/")) {
-			serverUrl = serverUrl.substring(0, serverUrl.length() - 1);
-		}
-		result.setType(CIServerTypes.JENKINS.value())
-				.setVersion(Jenkins.VERSION)
-				.setUrl(serverUrl)
-				.setSendingTime(System.currentTimeMillis());
-		return result;
-	}
-
-	public static void publishEventToRelevantClients(CIEvent event) {
-		OctaneSDK.getClients().forEach(c->c.getEventsService().publishEvent(event));
-	}
-
-	@Override
-	public String getParentJobName(String jobId) {
-		if (jobId != null && jobId.contains(BuildHandlerUtils.JOB_LEVEL_SEPARATOR)) {
-			int index = jobId.lastIndexOf(BuildHandlerUtils.JOB_LEVEL_SEPARATOR);
-			String parent = jobId.substring(0, index);
-			return parent;
-		}
-		return null;
-	}
-}
+    private PipelineNode createPipelineNodeFromJobName(String name) {
+        String[] defaultBranchesArray = config.getDefaultBranches().split(DEFAULT_BRANCHES_SEPARATOR);
+        Set<String> defaultBranches = Arrays.stream(defaultBranchesArray)
+                .map(String::trim)
+                .filter(StringUtils::isNotEmpty)
+                .collect(Collectors.toSet());
+
+        Item item = Jenkins.get().getItemByFullName(name);
+        Collection<? extends Job> allJobs = item.getAllJobs();
+
+        Job job = allJobs.stream()
+                .filter(tempJob -> defaultBranches.contains(getDisplayNameFromJob(tempJob)))
+                .findFirst().orElse(null);
+
+        PipelineNode result = dtoFactory.newDTO(PipelineNode.class)
+                .setJobCiId(BuildHandlerUtils.translateFolderJobName(name))
+                .setName(name);
+
+        if(job != null) {
+            String defaultBranch = getDisplayNameFromJob(job);
+            result.setParameters(ParameterProcessors.getConfigs(job))
```

Comment:
why do need to add the parameters? I didn't see on the original code?

Reply:
Actually, we do all those changes to get parameters from the default branch. I moved the logic for getting parameters to another method and it is only for multibranch jobs.